internal/notification/handlers/user_notifier.go

```diff
@@ -99,14 +102,16 @@ func NewUserNotifier(
 	channels types.ChannelChains,
 	otpEmailTmpl string,
 	legacyMode bool,
+	queue Queue,
 ) *handler.Handler {
 	if legacyMode {
 		return NewUserNotifierLegacy(ctx, config, commands, queries, channels, otpEmailTmpl)
 	}
 	return handler.NewHandler(ctx, &config, &userNotifier{
-		commands:     commands,
 		queries:      queries,
 		otpEmailTmpl: otpEmailTmpl,
+		queue:        queue,
+		maxAttempts:  config.MaxFailureCount,
```

Comment:
I think we can get the correct MaxAttempts by injecting the worker config into `NewUserNotifier`. But it's kinda ick. Maybe if we just called it `NotificationQueueConfig` instead that would make it more palatable? 🙂